src/prolly/map.rs

```diff
@@ -116,6 +125,75 @@ impl Map {
         self.pending.clear();
         Ok(self.base.as_ref().unwrap().chunk().hash().into())
     }
+
+    // Returns the diff between the pending entries and the already flushed entries.
+    pub fn pending_diff(&self) -> Result<Vec<String>, FromUtf8Error> {
+        let mut keys = Vec::with_capacity(self.pending.len());
+        for (key, pending_val) in self.pending.iter() {
+            match pending_val {
+                Some(pending_val) => match self.base_get(key) {
+                    Some(base_val) => {
+                        if pending_val != base_val {
+                            keys.push(String::from_utf8(key.clone())?);
```

Comment:
Needs to be copied later so I copied it earlier since it made the code simpler. I can think a bit more about this to see if it makes sense to go `[u8]` all the way to JS

Reply:
Going to do this in another PR